ui/main/src/app/business/view/processmonitoring/processmonitoring.view.ts

```diff
@@ -103,4 +104,191 @@ export class ProcessMonitoringView {
+                          '-' +
+                          String(activeToDateFormat.getMonth() + 2).padStart(2, '0') +
+                          '-01T00:00'
+            };
+        } else {
+            return {
+                activeFrom:
+                    activeFromDateFormat.getMonth() === 0
+                        ? activeFromDateFormat.getFullYear() - 1 + '-12-01T00:00'
+                        : activeFromDateFormat.getFullYear() +
+                          '-' +
+                          String(activeFromDateFormat.getMonth()).padStart(2, '0') +
+                          '-01T00:00',
+                activeTo:
+                    activeToDateFormat.getMonth() === 0
+                        ? activeToDateFormat.getFullYear() - 1 + '-12-01T00:00'
+                        : activeToDateFormat.getFullYear() +
+                          '-' +
+                          String(activeToDateFormat.getMonth()).padStart(2, '0') +
+                          '-01T00:00'
+            };
+        }
+    }
+
+    public getDatesWhenMovingWithWeekPeriodClicked(
+        activeFrom: string,
+        activeTo: string,
+        isForward: boolean
+    ): {activeFrom: any; activeTo: any} {
+        if (isForward) {
+            return {
+                activeFrom: moment(activeFrom, 'YYYY-MM-DDTHH:mm').add(7, 'days').format('YYYY-MM-DDTHH:mm'),
+                activeTo: moment(activeTo, 'YYYY-MM-DDTHH:mm').add(7, 'days').format('YYYY-MM-DDTHH:mm')
+            };
+        } else {
+            return {
+                activeFrom: moment(activeFrom, 'YYYY-MM-DDTHH:mm').subtract(7, 'days').format('YYYY-MM-DDTHH:mm'),
+                activeTo: moment(activeTo, 'YYYY-MM-DDTHH:mm').subtract(7, 'days').format('YYYY-MM-DDTHH:mm')
+            };
+        }
+    }
+
+    public getDatesWhenMoving(
+        activeFrom: string,
+        activeTo: string,
+        isForward: boolean
+    ): {activeFrom: any; activeTo: any} {
+        const activeFromDateFormat = new Date(activeFrom);
+        const activeToDateFormat = new Date(activeTo);
+
+        let newActiveFromDate: string, newActiveToDate: string;
+        if (isForward) {
+            newActiveFromDate = new Date(
+                activeFromDateFormat.getTime() + (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+            newActiveToDate = new Date(
+                activeToDateFormat.getTime() + (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+        } else {
+            newActiveFromDate = new Date(
+                activeFromDateFormat.getTime() - (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
+            newActiveToDate = new Date(
+                activeToDateFormat.getTime() - (activeToDateFormat.getTime() - activeFromDateFormat.getTime())
+            ).toLocaleString('fr-FR');
```

Comment:
Passing via the fr-FR format seems a little weird 

Could you just keep in this method a date object and in formatDateLocalStringForInput directly convert to the wanted format using for example moment or getYear(), get Month() ... to be consistant with other methods ?

Reply:
Done.